app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -527,6 +533,70 @@ class LuciBuildService {
     return buildBucketClient.getBuild(request);
   }
 
+  /// Schedules list of post-submit builds deferring work to [schedulePostsubmitBuild].
+  Future<void> schedulePostsubmitBuilds({
+    required Commit commit,
+    required List<Pair<Target, Task>> toBeScheduled,
+  }) async {
+    if (toBeScheduled.isEmpty) {
+      log.fine('Skipping schedulePostsubmitBuilds as there are no targets to be scheduled by Cocoon');
+      return;
+    }
+    final List<Request> buildRequests = <Request>[];
+    for (Pair<Target, Task> pair in toBeScheduled) {
+      final ScheduleBuildRequest scheduleBuildRequest =
+          _createPostsubmitScheduleBuild(commit: commit, target: pair.first, task: pair.second);
+      buildRequests.add(Request(scheduleBuild: scheduleBuildRequest));
+    }
+    final BatchRequest batchRequest = BatchRequest(requests: buildRequests);
+    log.fine(batchRequest);
+    await pubsub.publish('scheduler-requests', batchRequest);
+    log.info('Published a request with ${buildRequests.length} builds');
+  }
+
+  /// Creates a [ScheduleBuildRequest] for [target] and [task] against [commit].
+  ScheduleBuildRequest _createPostsubmitScheduleBuild({
+    required Commit commit,
+    required Target target,
+    required Task task,
+  }) {
+    final Map<String, List<String>> tags = <String, List<String>>{
+      'buildset': <String>[
+        'commit/git/${commit.sha}',
```

Comment:
should we have some helper functions to generate these different git related strings?

Reply:
I want to overall simplify `LuciBuildService` (there's a lot of redundancy within it that I want to move into Scheduler). I filed https://github.com/flutter/flutter/issues/97687 for tracking this.

My assumption is it will be more obvious to me in the coming weeks as I work on this as to what the right way is to de-duplicate this service.